src/service/errors/responses.go

```diff
@@ -1,7 +1,22 @@
 package errors
 
-import "github.com/hashicorp/vault/sdk/logical"
+import (
+	"github.com/ConsenSys/orchestrate-hashicorp-vault-plugin/src/pkg/errors"
+	"github.com/hashicorp/vault/sdk/logical"
+	"net/http"
+)
 
-func BadRequestError(req *logical.Request, msg string) (*logical.Response, error) {
-	return logical.RespondWithStatusCode(logical.ErrorResponse(msg), req, 400)
+func WriteHTTPError(req *logical.Request, err error) (*logical.Response, error) {
```

Comment:
The name of this function appers to be bit confusing. because it does not write a http response, it doess return one.

in addition this method appear to be doing two things, one building a response obj, then parsing an error. what about doing two separate methods?

Reply:
waiting for answer on this, going to be applied on next PR.